src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,160 @@
+    public static final String NEWLINE = System.lineSeparator();
+    public static final String INDENTATION = "    ";
+
+    /**
+     * Accepts user input from stdin.
+     *
+     * @param inputReader used to read user input
+     * @return String representing a user command
+     * @throws IOException when BufferedReader fails to read user input
+     */
+    protected String readCommand(BufferedReader inputReader) throws IOException {
+        String command;
+        command = inputReader.readLine().trim();
+        return command;
+    }
+
+    /**
+     * Prints given text with indentation: specified in duke.Duke class.
+     *
+     * @param text used for formatting and printing
+     */
+    protected void printTextWithIndentation(String text) {
+        System.out.println(INDENTATION + text);
+    }
+
+    /**
+     * Prints greeting message.
+     */
+    protected void greet() {
+        String logo = "  __  __        _____       _           _   " + NEWLINE
+                + " |  \\/  |      |  __ \\     | |         | |  " + NEWLINE
+                + " | \\  / |_ __  | |__) |___ | |__   ___ | |_ " + NEWLINE
+                + " | |\\/| | '__| |  _  // _ \\| '_ \\ / _ \\| __|" + NEWLINE
+                + " | |  | | |    | | \\ \\ (_) | |_) | (_) | |_ " + NEWLINE
+                + " |_|  |_|_|    |_|  \\_\\___/|_.__/ \\___/ \\__|";
+        System.out.println(logo);
+        printTextWithIndentation(HORIZONTAL_BAR);
+        printTextWithIndentation("Hello friend. Hello friend?");
+        printTextWithIndentation("That's lame. Maybe I should give you a name.");
+        printTextWithIndentation("But that's a slippery slope, you're only in my head,");
+        printTextWithIndentation("we have to remember that.");
+        printTextWithIndentation(HORIZONTAL_BAR);
+    }
+
+    /**
+     * Prints out error when storage data cannot be loaded into program.
+     */
+    public void showLoadingError() {
+        printTextWithIndentation(HORIZONTAL_BAR);
+        printTextWithIndentation("Unable to load storage data...");
+        printTextWithIndentation(HORIZONTAL_BAR);
+    }
+
+    /**
+     * Prints out error message for invalid user input.
+     */
+    public void unableToReadUserInput() {
+        printTextWithIndentation(HORIZONTAL_BAR);
+        printTextWithIndentation("Unable to read user input...");
+        printTextWithIndentation(HORIZONTAL_BAR);
+    }
+
+    /**
+     * Prints out empty command error.
+     */
+    public void commandNotFound() {
+        printTextWithIndentation(HORIZONTAL_BAR);
+        printTextWithIndentation("404 Not Found... Are you there?");
+        printTextWithIndentation(HORIZONTAL_BAR);
+    }
+
+    /**
+     * Prints out all tasks present in given duke.task.TaskList.
+     *
+     * @param tasks for printing task information
+     * @param header contains list information
+     */
+    public void listTasks(TaskList tasks, String header) {
```

Comment:
I feel that the logic of what to list should not be handled by the UI but rather by the Command itself. In general I think the UI should not have to know about Tasks directly, but rather just deal with user input and output. This might result in significant refactoring though (as there are other methods below that require UI to know about Tasks) so it might not be a necessary change.